From e1e68ef13d1bb81d2b383dc243c2b076f749717c Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Fri, 20 Aug 2021 11:56:30 +0530
Subject: [PATCH 8/9] coresight: Enable hardware quirks based on AMBA ID

commit 6bd549faaf953721932d17181f78e48352c3dfcd from
git@git.assembla.com:cavium/WindRiver.linux.git

Hardcoded Silicon ids removed for finding hardware quirks.

Change-Id: I7601c4c530ba592d0ff33a7f0b5814285882ff08
Signed-off-by: Linu Cherian <lcherian@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/60084
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/hwtracing/coresight/coresight-etm4x.c | 2 +-
 drivers/hwtracing/coresight/coresight-tmc.c   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/hwtracing/coresight/coresight-etm4x.c b/drivers/hwtracing/coresight/coresight-etm4x.c
index c91b2ccffce1..5b92a5e66df2 100644
--- a/drivers/hwtracing/coresight/coresight-etm4x.c
+++ b/drivers/hwtracing/coresight/coresight-etm4x.c
@@ -1237,7 +1237,7 @@ static int etm4_probe(struct amba_device *adev, const struct amba_id *id)
 	etm4_set_default(&drvdata->config);
 
 	/* Enable fixes for Silicon issues */
-	drvdata->etm_options = coresight_get_etm_quirks(OCTEONTX_CN9XXX_ETM);
+	drvdata->etm_options = coresight_get_etm_quirks(id->id);
 
 	desc.type = CORESIGHT_DEV_TYPE_SOURCE;
 	desc.subtype.source_subtype = CORESIGHT_DEV_SUBTYPE_SOURCE_PROC;
diff --git a/drivers/hwtracing/coresight/coresight-tmc.c b/drivers/hwtracing/coresight/coresight-tmc.c
index 8767afd01aae..941830b16552 100644
--- a/drivers/hwtracing/coresight/coresight-tmc.c
+++ b/drivers/hwtracing/coresight/coresight-tmc.c
@@ -469,7 +469,7 @@ static int tmc_probe(struct amba_device *adev, const struct amba_id *id)
 	drvdata->cpu = pdata ? pdata->cpu : 0;
 
 	/* Enable fixes for Silicon issues */
-	drvdata->etr_options = coresight_get_etr_quirks(OCTEONTX_CN9XXX_ETR);
+	drvdata->etr_options = coresight_get_etr_quirks(id->id);
 
 	/* Update the smp target cpu */
 	drvdata->rc_cpu = is_etm_sync_mode_sw_global() ? SYNC_GLOBAL_CORE :
-- 
2.31.1

